lib/common/util/user.js

```diff
@@ -2,15 +2,56 @@
 
 import objectPath from 'object-path'
 
-import type {Profile} from '../../types'
+import type { AccountType, DatatoolsSettings, Profile, UserProfile } from '../../types'
+import type { AppState } from '../../types/reducers'
+
+/**
+ * Predicate that determines whether a settings object correspond to the configured Auth0 client.
+ */
+export const isSettingForThisClient = (settings: DatatoolsSettings) => {
+  const clientId = process.env.AUTH0_CLIENT_ID
+  if (!clientId) throw new Error('Auth0 client ID must be set in config')
+
+  return settings.client_id === clientId
+}
+
+/**
+ * Obtains the entry for the user's app_metadata.datatools (in Auth0) that corresponds to the configured client,
+ * or null if no such entry exists.
+ */
+export const getSettingsForThisClient = (allClientSettings?: ?Array<DatatoolsSettings>): ?DatatoolsSettings =>
+  allClientSettings && allClientSettings.find(isSettingForThisClient)
+
+/**
+ * Variant of above method that takes in a UserProfile object.
+ */
+export const getSettingsFromProfile = (profile: ?UserProfile): ?DatatoolsSettings =>
+  profile &&
+  profile.app_metadata &&
+  getSettingsForThisClient(profile.app_metadata.datatools)
 
 export const getUserMetadataProperty = (
   profile: ?Profile,
   propertyString: string
 ) => {
-  const CLIENT_ID = process.env.AUTH0_CLIENT_ID
   const datatools = objectPath.get(profile, 'user_metadata.datatools')
-  const application =
-    datatools && datatools.find(d => d.client_id === CLIENT_ID)
+  const application = getSettingsForThisClient(datatools)
   return objectPath.get(application, propertyString)
 }
+
+/**
+ * Gets the configured account types
+ */
+export function getAccountTypes (state: AppState): Array<AccountType> {
+  const { appInfo } = state.status
+  const { licensing } = (appInfo && appInfo.config.modules) || {}
+  return (licensing && licensing.enabled && licensing.account_types) || []
+}
+
+/**
+ * Gets the default account type, if any.
+ */
+export function getDefaultAccountType (accountTypes: Array<AccountType>): ?AccountType {
```

Comment:
Could the "default account" be moved up one level in the config so that it's not possible to specify more than one default? Or would it be too difficult to get that config item to this function?

Reply:
Good point, which leads me to question why I picked an array instead of a map to list account types.

Reply:
Thanks for this very good suggestion. Commit https://github.com/ibi-group/datatools-ui/pull/855/commits/406555e72d66df5c782f929bc11c9e61e50642f4 turns `accountTypes` into a map, and the default account type is now denoted with the `default` key, as in below. The `default` key is used in code to seek whether a default account has been set. Let me know what you think @daniel-heppner-ibigroup and @miles-grant-ibigroup:
```
modules:
  licensing:
    account_types:
      default:
        name: Standard
        require_terms: false
      trial-period:
        name: Trial Account
        require_terms: true
        # TODO: provide actual terms URL for free accounts
        terms_url: https://mit-license.org/
```

Reply:
This is clever! I like it.

Reply:
I like it too! Clever adaptation, I didn't think of this either when I wrote my comment!